src/main/java/frc/robot/vision/sources/VisionSource.java

```diff
@@ -0,0 +1,18 @@
+package frc.robot.vision.sources;
+
+import frc.robot.vision.data.VisionData;
+import frc.utils.Filter;
+
+import java.util.Optional;
+
+public interface VisionSource<ReturnType extends VisionData> {
+
```

Comment:
[...](https://github.com/GreenBlitz/GB-Robot-Template/pull/501/files#r1900179053)
and fore multivisionsources too